src/cuda/statistics.cu

```diff
@@ -449,11 +491,11 @@ namespace vcuda{
                   hist[4 * i + 0] = mx * imm;
                   hist[4 * i + 1] = my * imm;
                   hist[4 * i + 2] = mz * imm;
-                  hist[4 * i + 3] = imm * ms;
+                  hist[4 * i + 3] = ms / imm;
```

Comment:
imm is actuall 1/|M|. Actually looking at it again this probably needs to be 1.0/( ms* imm)